csharp/src/Ice/MetricsObserverI.cs

```diff
@@ -126,6 +129,31 @@ override protected object resolve(object obj)
             readonly private System.Reflection.FieldInfo _field;
         }
 
+        private class MemberPropertyResolverI : Resolver
```

Comment:
I don't find this code elegant at all. However, it's consistent with what we do currently.